integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/APIKeyHeaderTestCase.java

```diff
@@ -55,24 +54,9 @@
     @BeforeClass(description = "Initialise the setup for API key tests")
     void start() throws Exception {
         super.initWithSuperTenant();
-        APIRequest apiRequest = new APIRequest(SAMPLE_API_NAME, SAMPLE_API_CONTEXT,
-                new URL(Utils.getDockerMockServiceURLHttp(TestConstant.MOCK_BACKEND_BASEPATH)));
-        apiRequest.setProvider(user.getUserName());
-        apiRequest.setVersion(SAMPLE_API_VERSION);
-        apiRequest.setTiersCollection(TestConstant.SUBSCRIPTION_TIER.UNLIMITED);
-        apiRequest.setTier(TestConstant.API_TIER.UNLIMITED);
-        apiRequest.setApiTier(TestConstant.API_TIER.UNLIMITED);
-
-        // Add api related operations
-        APIOperationsDTO findByStatus = new APIOperationsDTO();
-        findByStatus.setVerb("GET");
-        findByStatus.setTarget("/pet/{petId}");
-        findByStatus.setThrottlingPolicy(TestConstant.API_TIER.UNLIMITED);
-
-        List<APIOperationsDTO> operationsDTOS = new ArrayList<>();
-        operationsDTOS.add(findByStatus);
-        apiRequest.setOperationsDTOS(operationsDTOS);
 
+        APIRequest apiRequest = PublisherUtils.createSampleAPIRequest(SAMPLE_API_NAME, SAMPLE_API_CONTEXT,
```

Comment:
Does this enable apiKey as an app level auth mechanism for this API?

Reply:
In `PublisherUtils.java` class there is a method to return API request object. https://github.com/wso2/product-microgateway/blob/main/integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/utils/PublisherUtils.java#L176
After calling that method, we can add `securitySchemesList` to it (similar to the previous implementation). After that, test cases follow same pervious approach. https://github.com/wso2/product-microgateway/blob/c49e2c555faefe1ccdc6ed6853cf0a76af9aacc9/integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/APIKeyHeaderTestCase.java#L61

Reason to do above change is, same code repeated in multiple test cases previously. By using `createSampleAPIRequest` method we can resolve that.